hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/pipeline/ClosePipelineSubcommand.java

```diff
@@ -38,14 +40,59 @@
   @CommandLine.ParentCommand
   private PipelineCommands parent;
 
-  @CommandLine.Parameters(description = "ID of the pipeline to close")
+  @CommandLine.Parameters(description = "ID of the pipeline to close,"
+      + "'ALL' means all pipeline")
   private String pipelineId;
 
+  @CommandLine.Option(names = {"-ffc", "--filterByFactor"},
+      description = "Filter listed pipelines by Factor(ONE/one)",
+      defaultValue = "",
+      required = false)
+  private String factor;
+
+  @CommandLine.Option(names = {"-fst", "--filterByState"},
+      description = "Filter listed pipelines by State(OPEN/CLOSE)",
+      defaultValue = "",
+      required = false)
+  private String state;
+
   @Override
   public Void call() throws Exception {
     try (ScmClient scmClient = parent.getParent().createScmClient()) {
-      scmClient.closePipeline(
-          HddsProtos.PipelineID.newBuilder().setId(pipelineId).build());
+      if (pipelineId.equalsIgnoreCase("ALL")) {
+        if (Strings.isNullOrEmpty(factor) && Strings.isNullOrEmpty(state)) {
+          scmClient.listPipelines().forEach(pipeline -> {
+            try {
+              scmClient.closePipeline(
+                  HddsProtos.PipelineID.newBuilder()
+                      .setId(pipeline.getId().getId().toString()).build());
+            } catch (IOException e) {
```

Comment:
Should we continue closing the remaining and print out a list of pipeline closed successfully and unsuccessfully at the end?